client/checkout/utils/upe.js

```diff
@@ -25,42 +30,23 @@ export const getTerms = ( paymentMethodsConfig, value = 'always' ) => {
 };
 
 /**
- * Finds selected payment gateway and returns matching Stripe payment method for gateway.
+ * Returns Stripe payment method (e.g. card, bancontact ) for selected payment gateway.
  *
- * @return {string} Stripe payment method type
+ * @return {string} Payment method name
  */
 export const getSelectedUPEGatewayPaymentMethod = () => {
-	const paymentMethodsConfig = getUPEConfig( 'paymentMethodsConfig' );
-	const gatewayCardId = getUPEConfig( 'gatewayId' );
-	let selectedGatewayId = null;
-
-	// Handle payment method selection on the Checkout page or Add Payment Method page where class names differ.
-	const radio = document.querySelector(
-		'li.wc_payment_method input.input-radio:checked, li.woocommerce-PaymentMethod input.input-radio:checked'
+	const selectedGateway = document.querySelector(
+		`.wcpay-upe-form[checked]`
```

Comment:
Thanks for mentioning the `checked` attribute purpose for other types of elements 👍 
Regarding the usage of input element from WooCoommerce - I knew about this element but my assumption was we want to avoid targeting them as much as possible, that's why I went with our custom one which forced me to create some handling of updating selected payment method (another comment from you below). 

Another option would be to traverse from our div to the checked input.
I'd love to hear your thoughts about targeting Core html elements, thank you!

Reply:
I understand the worry about saving ourselves from potential changes from WooCommerce Core's elements or structure, which could make our implementation brittle.

However, the selector `.woocommerce-checkout input[name="payment_method"]:checked` (which targets WooCommerce Core elements) is already used in this implementation in the `payment_method_selected` listener.
I don't think that further wrapping this logic in custom handling necessarily isolates the code more effectively.

Traversing the `.wcpay-upe-form` to the `input` could be a middle-ground, but I don't think it makes it less brittle.

I was recommending using the `'input[name="payment_method"][value*="woocommerce_payments"]:checked'` selector because the `payment_method` inputs are indeed rendered by WC Core, but WC Core also relies on them to process the shortcode checkout.
It's unlikely for a theme to interfere with them.
Since there can only be one `input[name="payment_method"]` element with the `checked` attribute, I thought it was safe to use this selector.

Please let me know if I missed anything! :)

Reply:
> However, the selector .woocommerce-checkout input[name="payment_method"]:checked (which targets WooCommerce Core elements) is already used in this implementation in the payment_method_selected listener

Well, that's actually a fair point. 

I removed the event handler and applied your suggestion in https://github.com/Automattic/woocommerce-payments/pull/9782/commits/e81e3a469071bcb109557de63e494e9529b1e8da.

I did that with all the comments, because they seem to be helpful. Also, I wanted to modify your suggestion by adding `card` instead of an empty string in the second replace, but upon analyzing it realized that it's actually less descriptive and in line with the first replace, so I kept things as they are while falling back to card in the return statement.